src/modules/35-connectors/components/CreateConnector/CreateAzureBlobConnector/views/AzureBlobForm.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { useState } from 'react'
+import * as Yup from 'yup'
+import { StepProps, Container, Text, Formik, FormikForm, Layout, Button, ButtonVariation } from '@wings-software/uicore'
+import { FontVariation } from '@harness/design-system'
+import { useStrings } from 'framework/strings'
+import { PageSpinner } from '@common/components'
+import { setupAzureBlobFormData } from '@connectors/pages/connectors/utils/ConnectorUtils'
+import type { SecretReference } from '@secrets/components/CreateOrSelectSecret/CreateOrSelectSecret'
+import type { StepDetailsProps, ConnectorDetailsProps } from '@connectors/interfaces/ConnectorInterface'
+import AzureBlobFormFields from './AzureBlobFormFields'
+import css from '../CreateAzureBlobConnector.module.scss'
+
+export interface AzureBlobFormData {
+  clientId?: string
+  secretKey?: SecretReference
+  tenantId?: string
+  containerURL?: string
+  default?: boolean
+}
+
+const AzureBlobForm: React.FC<StepProps<StepDetailsProps> & ConnectorDetailsProps> = props => {
+  const { prevStepData, previousStep, isEditMode, nextStep, connectorInfo, accountId } = props
+  const { getString } = useStrings()
+
+  const defaultInitialFormData: AzureBlobFormData = {
+    clientId: undefined,
+    tenantId: undefined,
+    containerURL: undefined,
+    secretKey: undefined,
+    default: false
+  }
+
+  const [initialValues, setInitialValues] = useState(defaultInitialFormData)
+  const [loadingFormData, setLoadingFormData] = useState(isEditMode)
+
+  React.useEffect(() => {
+    if (isEditMode && connectorInfo) {
+      setupAzureBlobFormData(connectorInfo, accountId).then(data => {
+        setInitialValues(data as AzureBlobFormData)
+        setLoadingFormData(false)
+      })
+    }
+  }, [isEditMode, connectorInfo])
```

Comment:
Codacy has a fix for the issue: [React Hook React.useEffect has a missing dependency: 'accountId'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232296)
```suggestion
  }, [isEditMode, connectorInfo, accountId])
```